numba/typeconv/typeconv.py

```diff
@@ -23,6 +23,9 @@ def select_overload(self, sig, overloads, allow_unsafe):
                                          allow_unsafe)
 
     def check_compatible(self, fromty, toty):
+        if not hasattr(toty, '_code'):
```

Comment:
Should we just do a isinstance() check against numba.types.Type?  I don't think we will support anything that is structurally equivalent to a Type object but not a subclass of it.

Reply:
I suppose that is likely and it's not onerous to change if we decide differently later. Will patch.